### PR TITLE
Improve the development environment

### PR DESCRIPTION
**Pull request by docjyJ**
- [x] Delete the unnecessary `.idea` folder
- [x] Added some commands to start AIO application development
- [x] Adding tools used in development requirements
- [x] Updated workflows that no longer need to install tools 
- [x] adjust .gitignore file

### CODE REVIEW

**Review by szaimen — COMMENTED** [review]
Thanks a lot for tackling this @docjyJ! :)

Some comments below:

**Comment by szaimen** [comment]
Btw, @docjyJ what about your changes to the gitignore file? https://github.com/nextcloud/all-in-one/pull/5016/files#diff-3928fe1f5e72c4886b92733a2978fab5c1b036c7fd23e652fe9dd049572d2940

File: php/composer.json [hunk 1]
```diff
@@ -18,10 +18,20 @@
...
+            "Composer\\Config::disableProcessTimeout",
+            "php -S localhost:8080 -t public"
+        ],
 		"psalm": "psalm --threads=1",
-		"psalm:update-baseline": "psalm --threads=1 --update-baseline",
-		"lint": "find . -name \\*.php -not -path './vendor/*' -exec php -l {} \\;",
-		"php-deprecation-detector": "find . -name \\*.php -not -path './vendor/*' -exec phpdd scan {} -n -t 8.3 \\;"
+		"psalm:update-baseline": "psalm --threads=1 --monochrome --no-progress --output-format=text --update-baseline",
+		"lint": "php -l src/*.php src/**/*.php public/index.php",
+        "lint:twig": "twig-linter lint ./templates",
```
Comment (szaimen) [inline php/composer.json:34]:
```suggestion
		"lint:twig": "twig-linter lint ./templates",
```

File: php/README.md [hunk 2]
```diff
@@ -7,10 +7,34 @@ This is the code for the PHP Docker controller.
...
+If you have an error that says `Couldn't connect to server for http://127.0.0.1/v1.41/networks/create` makes sure a `nextcloud-aio-mastercontainer` is running.
+You can start it with the following command:
+
+```bash
+sudo docker run \
+--rm \
+--name nextcloud-aio-mastercontainer \
+--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
+--volume /var/run/docker.sock:/var/run/docker.sock:ro \
+nextcloud/all-in-one:latest
```
[the remark this answers is not in the document]
Comment (docjyJ) [reply php/README.md:26]: I tried but the MasterContainer is looking for its docker configuration, for the updates...
I'll try again and send you the error it gives.

File: php/README.md [hunk 3]
```diff
@@ -7,10 +7,34 @@ This is the code for the PHP Docker controller.
 Running this locally requires Docker Engine on the same machine. 
 If this is the case, just execute the following command:
 
-```
-composer install --no-dev
-cd public/
-php -S 0.0.0.0:8080
+```bash
+composer install
+sudo SKIP_DOMAIN_VALIDATION=true composer run dev # sudo is required to access docker socket
```
[the remark this answers is not in the document]
Comment (docjyJ) [reply php/README.md:12]: Agree

File: php/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (szaimen) [reply php/README.md:26]: ah I see. Yeah then you are probably right that we need the mastercontainer...

**Comment by docjyJ** [comment]
> Btw, @docjyJ what about your changes to the gitignore file? https://github.com/nextcloud/all-in-one/pull/5016/files#diff-3928fe1f5e72c4886b92733a2978fab5c1b036c7fd23e652fe9dd049572d2940
```.gitignore
data/*
session/*
!data/.gitkeep
!session/.gitkeep
```
This keeps the directory but prevents committing all session files created at runtime.
These files don't have a template to blacklist them, so I whitelist the `.gitkeep`.

The `vendor` is like `node_module` for composer, it must be in gitigniore.

And `.idea`, I don't know why it was committed...

**Comment by szaimen** [comment]
> > Btw, @docjyJ what about your changes to the gitignore file? https://github.com/nextcloud/all-in-one/pull/5016/files#diff-3928fe1f5e72c4886b92733a2978fab5c1b036c7fd23e652fe9dd049572d2940
> 
> ```gitignore
> data/*
> session/*
> !data/.gitkeep
> !session/.gitkeep
> ```
> 
> This keeps the directory but prevents committing all session files created at runtime. These files don't have a template to blacklist them, so I whitelist the `.gitkeep`.
> 
> The `vendor` is like `node_module` for composer, it must be in gitigniore.
> 
> And `.idea`, I don't know why it was committed...

sounds good. I'd recommend to commit the changes to the gitignore file to this branch :)

**Comment by docjyJ** [comment]
> > > Btw, @docjyJ what about your changes to the gitignore file? https://github.com/nextcloud/all-in-one/pull/5016/files#diff-3928fe1f5e72c4886b92733a2978fab5c1b036c7fd23e652fe9dd049572d2940
> > 
> > ```gitignore
> > data/*
> > session/*
> > !data/.gitkeep
> > !session/.gitkeep
> > ```
> > 
> > This keeps the directory but prevents committing all session files created at runtime. These files don't have a template to blacklist them, so I whitelist the `.gitkeep`.
> > 
> > The `vendor` is like `node_module` for composer, it must be in gitigniore.
> > 
> > And `.idea`, I don't know why it was committed...
> 
> sounds good. I'd recommend to commit the changes to the gitignore file to this branch :) 

Yes, I'll split into two branches.

**Comment by docjyJ** [comment]
My bad, I forgot to commit the `.gitignore`.

File: php/composer.lock [hunk 4]
```diff
@@ -4,20 +4,20 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "4dcdd3b6df3f2041895d4db74bd45102",
+    "content-hash": "0e1d24f3fa776163acefdebc91da39d3",
     "packages": [
         {
             "name": "guzzlehttp/guzzle",
-            "version": "7.9.2",
+            "version": "7.9.0",
```
[the remark this answers is not in the document]
Comment (docjyJ) [reply php/composer.lock]: 5dbcf17

File: php/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (docjyJ) [reply php/README.md:26]: image

I confirm that it requires an update...

[the remark this answers is not in the document]
Comment (szaimen) [reply php/README.md:26]: Ok

File: php/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (docjyJ) [reply php/README.md]: 68b5b770

**Review by szaimen — APPROVED** [review]
LGTM. Thanks a lot! :)

**Pull request merged**